src/canvasObject.js

```diff
@@ -1,159 +1,128 @@
   this.opacity = config.opacity || 1;
-  this.x = config.x || 0;
-  this.y = config.y || 0;
-  this.placeholder = config.placeholder || { type: 'image', url: './example-thumbnail.png' };
   this.index = config.index;
 
   this.id = config.canvas['@id'];
-  this.height = config.canvas.height;
-  this.width = config.canvas.width;
-  this.images = config.canvas.images;
+
+  this.bounds = {
+    x : config.x || 0,
+    y : config.y || 0,
+    height : config.canvas.height,
+    width : config.canvas.width
+  };
+  this.thumbUrl = config.canvas.thumbnail;
+  this.thumbService = config.canvas.images[0].resource.service['@id'];
+
+  // details and alternates possibly go here; disambiguate between them.
+  this.images = config.canvas.images.map(function(image) {
+    return new ImageResource(
+      {
+        tileSource: image.resource.service['@id'] + '/info.json'
+      },
+      self,
+      dispatcher
+    );
+  });
+
   this.label = config.canvas.label;
-  this.tileSourceUrl = config.canvas.images[0].resource.service['@id'] + '/info.json';
   this.viewingHint = config.canvas.viewingHint;
 
   this.dispatcher = dispatcher;
 };
 
 CanvasObject.prototype = {
-  openTileSource: function(viewer) {
-    var self = this;
+  openTileSource: function(viewer, imageIndex) {
+    this.dispatcher.emit('detail-tile-source-opened', { 'detail': this.id });
+    var thumbnail = this.thumbnail;
+    var image = this.images[imageIndex];
+
+    var onTileDrawn = function(event) {
+      if(event.detail === image.tileSource) {
+        image.hide(true);
+        image.fade(1);
+
+        if(thumbnail){
+          thumbnail.destroy(viewer);
+        }
+      }
+    };
 
-    // We've already loaded this tilesource instead of the thumbnail
-    if(this.fullyOpened) {
-      return;
-    }
-
-    // otherwise, continue loading the tileSource.
-    this.dispatcher.emit('detail-tile-source-requested', { 'detail': this.id });
-    viewer.addTiledImage({
-      x: this.x,
-      y: this.y,
-      width: this.width,
-      tileSource: this.tileSourceUrl,
-      opacity: this.opacity,
-      clip: this.clipRegion,
-      index: 0, // Add the new image below the stand-in.
-
-      success: function(event) {
-        var main = event.item;
-        self.fullyOpened = true;
-
-        var tileDrawnHandler = function(event) {
-          if (event.tiledImage === main) {
-            var previousImageObj = self._mainImageObj;
-
-            viewer.removeHandler('tile-drawn', tileDrawnHandler);
-            main.setOpacity(0, true);
-            self._fade(main, 1);
-            self._setMainImage(main);
-
-            if(previousImageObj){
-              viewer.world.removeItem(previousImageObj);
-            }
-            self.dispatcher.emit('detail-tile-source-opened', { 'detail': self.id });
-          }
-        };
-        viewer.addHandler('tile-drawn', tileDrawnHandler);
-      },
+    this.dispatcher.once('image-resource-tile-source-opened', onTileDrawn);
```

Comment:
Using `once` here might get us in trouble if a bunch of tile sources are being opened all at once. One option would be to change it to `on` and then do an `off` when you determine you have found the right event. Another option would be to pass a callback into ImageResource.openTileSource.